### ISSUE TITLE
have_attributes/respond_to matchers don't work with ActiveRecord

### ISSUE THREAD

**Issue by dolsem** [rspec/rspec-expectations#1162]
### Subject of the issue

I am trying to use the `have_attributes` matcher on a Rails ActiveRecord instance. It fails with the undefined method error. The issue is with the `respond_to` matcher that `have_attributes` is using under the hood. Specifically, I've tracked it down to this line: https://github.com/rspec/rspec-support/blob/9cec8e858228ee1d1395e4e6186ac39c2a83d705/lib/rspec/support.rb#L51
The problem is that ActiveRecord attributes are not regular methods, so `.method(:attribute_name)` fails. However, both `.send(:attribute_name)` and `.respond_to?(:attribute_name)` work just fine, which is why I consider this a bug, since the docs state that:
```ruby
expect(obj).to respond_to(:foo) # pass if obj.respond_to?(:foo)
```

This seems similar to what was happening in #1107.

### Your environment
* Ruby version: 2.5.5p157
* Rails version: 5.0.7.2
* rspec-expectations version: 3.8.4

### Steps to reproduce
```ruby
expect(active_record_instance).to respond_to :attribute_name
```

### Expected behavior
Expectation passes if the ActiveRecord model has an attribute called `attribute_name` (i.e. `active_record_instance.responds_to?(:attribute_name)` returns `true`)

### Actual behavior
`binding.pry` before [the culrpit line](https://github.com/rspec/rspec-support/blob/9cec8e858228ee1d1395e4e6186ac39c2a83d705/lib/rspec/support.rb#L51):
```
[14] pry(RSpec::Support)> KERNEL_METHOD_METHOD.bind(object).call(method_name)
NameError: undefined method `total' for class `#<Class:0x00000001133d0fe0>'
from (pry):20:in `method'
[15] pry(RSpec::Support)> object.__send__(:total)
=> 0
[16] pry(RSpec::Support)> KERNEL_METHOD_METHOD.bind(object).call(:total)
NameError: undefined method `total' for class `#<Class:0x00000001133d0fe0>'
from (pry):22:in `method'
[17] pry(RSpec::Support)> object.respond_to? :total
=> true
```

**Comment by dolsem** [comment]
Can `.method` on line 51 be changed to `.respond_to?`?

**Comment by dolsem** [comment]
I propose rescuing (`method_signature_for` invocation)[          Support::StrictSignatureVerifier.new(method_signature_for(actual, name))] and returning the result of `.responds_to?` in the special case of @expected_arity being zero.

**Comment by JonRowe** [comment]
That won't work, you need to rescue the error in `matches_arity?` and return true when there is no expected arity, and raise an argument error similar to #1107 saying that as the method doesn't support method you can't place an expectation on arity.

**Comment by JonRowe** [comment]
This was slightly more complex than original indicated:
```
expect(active_record_instance).to respond_to(:attribute_name)
```
Passes because there is no arity check, if you added an arity check it would then fail as described... However the `have_attributes` matcher always check that an attribute method has 0 arguments to avoid false positives. So I can see why the OP had issues (there are scenarios where rails defines the methods in reality which makes this even more complicated when used with rspec-rails, but the solution can ignore that).

Closing because this is fixed by #1164 which turns the error into a real warning when encountered on `respond_to` with instructions on how to resolve it, and by ignoring that new warning for `have_attributes`.
